ui/push-health/Metric.jsx

```diff
@@ -3,12 +3,31 @@ import PropTypes from 'prop-types';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faMinusSquare } from '@fortawesome/free-regular-svg-icons';
 import { Button, Badge, Row, Col, Collapse, Card, CardBody } from 'reactstrap';
+import camelCase from 'lodash/camelCase';
 
 import { resultColorMap } from './helpers';
 
 export default class Metric extends React.PureComponent {
+  componentDidUpdate(prevProps) {
+    const { name, expanded } = this.props;
+    const metricName = `${camelCase(name)}Metric`;
+
+    if (expanded && !prevProps.expanded) {
+      const elem = document.getElementById(metricName);
+
+      if (elem) {
+        // componentDidUpdate happens after the props are updated, but prior
+        // to rendering.  So we must wait just a bit before the elements will
+        // exist and we try to scroll to them.
+        setTimeout(() => {
+          elem.scrollIntoView({ behavior: 'smooth', block: 'center' });
```

Comment:
There's a `scrollToLine` function in utils file that work here.

Reply:
Oh cool.  OK, used that.  I modified the function to take `options` for smooth scrolling and centering.  I tested the LogViewer to make sure I didn't break it, though.  :)